src/Plugins/ComplexCore/src/ComplexCore/Filters/CropImageGeometry.cpp

```diff
@@ -0,0 +1,492 @@
+  }
+  if(yMax < yMin)
+  {
+    std::string ss = fmt::format("Y Max ({}) less than Y Min ({})", yMax, yMin);
+    return {MakeErrorResult<OutputActions>(-5550, ss)};
+  }
+  if(zMax < zMin)
+  {
+    std::string ss = fmt::format("Z Max ({}) less than Z Min ({})", zMax, zMin);
+    return {MakeErrorResult<OutputActions>(-5550, ss)};
+  }
+  if(xMin < 0)
+  {
+    std::string ss = fmt::format("X Min ({}) less than 0", xMin);
+    return {MakeErrorResult<OutputActions>(-5550, ss)};
+  }
+  if(yMin < 0)
+  {
+    std::string ss = fmt::format("Y Min ({}) less than 0", yMin);
+    return {MakeErrorResult<OutputActions>(-5550, ss)};
+  }
+  if(zMin < 0)
+  {
+    std::string ss = fmt::format("Z Min ({}) less than 0", zMin);
+    return {MakeErrorResult<OutputActions>(-5550, ss)};
+  }
+
+  // Validate the incoming DataContainer, Geometry, and AttributeMatrix ; bail if any do not exist since we plan on using them later on in the dataCheck
+  auto oldDimensions = getCurrentVolumeDataContainerDimensions(data, srcImagePath);
+  auto oldResolution = getCurrentVolumeDataContainerResolutions(data, srcImagePath);
+
+  auto srcImageGeom = data.getDataAs<ImageGeom>(srcImagePath);
+  auto oldOrigin = srcImageGeom->getOrigin();
+
+  std::vector<usize> tDims(3, 0);
+  if(xMax - xMin < 0)
+  {
+    xMax = xMin + 1;
+  }
+  if(yMax - yMin < 0)
+  {
+    yMax = yMin + 1;
+  }
+  if(zMax - zMin < 0)
+  {
+    zMax = zMin + 1;
+  }
+
+  tDims[0] = (xMax - xMin) + 1;
+  tDims[1] = (yMax - yMin) + 1;
+  tDims[2] = (zMax - zMin) + 1;
+
+  IntVec3 newDimensions;
+  newDimensions[0] = tDims[0];
+  newDimensions[1] = tDims[1];
+  newDimensions[2] = tDims[2];
+
+  FloatVec3 newResolution;
+  newResolution = oldResolution;
+
+  std::vector<float32> newOrigin(3);
+  if(shouldUpdateOrigin)
+  {
+    newOrigin[0] = xMin * newResolution[0] + oldOrigin[0];
+    newOrigin[1] = yMin * newResolution[1] + oldOrigin[1];
+    newOrigin[2] = zMin * newResolution[2] + oldOrigin[2];
+  }
+  else
+  {
+    newOrigin[0] = oldOrigin[0];
+    newOrigin[1] = oldOrigin[1];
+    newOrigin[2] = oldOrigin[2];
+  }
+
+  // saveAsNewImage
+  {
+    int64 XP = ((xMax - xMin) + 1);
+    int64 YP = ((yMax - yMin) + 1);
+    int64 ZP = ((zMax - zMin) + 1);
+
+    std::vector<usize> tDims(3, 0);
+    tDims[0] = XP;
+    tDims[1] = YP;
+    tDims[2] = ZP;
+
+    auto spacing = srcImageGeom->getSpacing();
+    std::vector<float32> spacingVec(3);
+    for(usize i = 0; i < 3; i++)
+    {
+      spacingVec[i] = spacing[i];
+    }
+    auto action = std::make_unique<CreateImageGeometryAction>(destImagePath, tDims, newOrigin, spacingVec);
+    actions.actions.push_back(std::move(action));
+
+    auto newCellFeaturesPath = destImagePath.createChildPath(newCellFeaturesName);
+    {
+      auto action = std::make_unique<CreateDataGroupAction>(newCellFeaturesPath);
+      actions.actions.push_back(std::move(action));
+    }
+
+    auto* srcImageGeom = data.getDataAs<ImageGeom>(srcImagePath);
```

Comment:
This line shadows the variable on line 235 and looks like it's getting the same geometry anyway.